rest/rest.go

```diff
@@ -51,6 +51,7 @@ type EndpointType client.EndpointType
 
 // Server contains configuration and handlers for additional listeners to be instantiated after app startup.
 type Server struct {
+	CoreAPI     bool
```

Comment:
Could you also add some validation here that ensures `Address`, `Protocol`, and `Certificate` are unset if `CoreAPI` is true?